pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -1033,8 +1105,7 @@ func TestListUsersExclusion(t *testing.T) {
 	})
 	tests := ListUsersTests{
```

Comment:
These tests look great, nice work! Can we add one or two more though that tests block lists through self-referntial cases, e.g.

```
model
  schema 1.1

type user

type group
  relations
    define member: [user, group#member] but not other
    define other: [user, group#member] or member
```

**Test Case 1**
- group:1#member@user:jon

`ListUsers(group:1#member, user_filters=[{"type": "user"}])` --> [] (should always return empty, no error)

**Test Case 2**
- group:2#member@group:1#member
- group:1#member@user:will

`ListUsers(group:2#member, user_filters=[{"type": "user"}])` --> [] (should always return empty, no error)

**Test Case 3**
```
model
  schema 1.1

type user

type group
  relations
    define member: [user, group#member] but not other
    define other: [user, group#member]
```
- group:1#other@group:1#member
- group:1#member@user:will

`ListUsers(group:1#member, user_filters=[{"type": "user"}])` --> [] (should always return empty, no error)

Reply:
Added case 3. Case 1 and 2 produced cycles.